bugs/gh-234.txt

```diff
@@ -0,0 +1,847 @@
+=== Begin SRU Template ===
+[Impact]
+Some cloud environments pass multipart user-data as text/x-shellscript.
+We need to make sure that it is accepted by cloud-init as user-data.
+
+[Test Case]
+
+```
+#!/bin/sh
+set -x
+# Manually deploy on a lxc container
+
+echo -e 'The userdata-mime file was generated by running the command:
+tools/make-mime.py --attach userdata:x-shellscript  > userdata-mime
+
+where userdata is defined as the following file:
+#cloud-config
+final_message: The system is finally up, after seconds $UPTIME seconds'
+
+cat > userdata-mime << EOF
+Content-Type: multipart/mixed; boundary="===============0816720877949849914=="
+MIME-Version: 1.0
+
+--===============0816720877949849914==
+Content-Type: text/x-shellscript; charset="utf-8"
+MIME-Version: 1.0
+Content-Transfer-Encoding: base64
+Content-Disposition: attachment; filename="userdata"
+
+I2Nsb3VkLWNvbmZpZwpmaW5hbF9tZXNzYWdlOiBUaGUgc3lzdGVtIGlzIGZpbmFsbHkgdXAsIGFm
+dGVyIHNlY29uZHMgJFVQVElNRSBzZWNvbmRzCg==
+
+--===============0816720877949849914==--
+EOF
+
+cat > setup_proposed.sh <<EOF
+#/bin/bash
+mirror=http://archive.ubuntu.com/ubuntu
+echo deb \$mirror \$(lsb_release -sc)-proposed main | tee /etc/apt/sources.list.d/proposed.list
+apt-get update -q;
+apt-get install -qy cloud-init;
+EOF
+
+recreate_issue() {
+   series=$1
+
+   name_recreate=test-$series-recreate
+   lxc delete $name_recreate --force 2> /dev/null
+   lxc launch ubuntu-daily:$series $name_recreate -c user.user-data="$(cat userdata-mime)"
+
+   lxc exec $name_recreate -- cloud-init status --wait --long
+   lxc exec $name_recreate -- cat /var/lib/cloud/seed/nocloud-net/user-data
```

Comment:
A note that you can dump userdata with the following command: `cloud-init query userdata`

Reply:
Got ir, I will update the code